### PR TITLE
stylo: Support system colors

### PR DESCRIPTION
**Pull request by Manishearth**
r=heycam from https://bugzilla.mozilla.org/show_bug.cgi?id=1340696

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15974)

### CODE REVIEW

File: components/style/properties/longhand/color.mako.rs [hunk 1]
```diff
@@ -53,4 +55,91 @@
...
+            pub fn parse(input: &mut Parser) -> Result<Self, ()> {
+                use std::ascii::AsciiExt;
+                static PARSE_ARRAY: &'static [(&'static str, SystemColor); ${len(system_colors)}] = &[
+                    % for color in system_colors:
+                        ("${color}", LookAndFeel_ColorID::eColorID_${to_rust_ident(color)}),
+                    % endfor
+                ];
+
+                let ident = input.expect_ident()?;
+                for &(name, color) in PARSE_ARRAY.iter() {
```
Comment (emilio) [inline components/style/properties/longhand/color.mako.rs]: don't we have a `case_insensitive_phf_map` thing just for this?

Comment (Manishearth) [inline components/style/properties/longhand/color.mako.rs]: Yes, but I'm not sure if it's worth it for a small list like this.

Comment (emilio) [inline components/style/properties/longhand/color.mako.rs]: I believe it's worth to be consistent. And `case_insensitive_phf_map` shouldn't be slower or anything.

Comment (Manishearth) [inline components/style/properties/longhand/color.mako.rs]: Works for me.

Comment (Manishearth) [inline components/style/properties/longhand/color.mako.rs]: filed https://github.com/servo/servo/issues/15994

**Pull request merged**